src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -89,13 +90,21 @@ withMiningCoordination logger conf cdb inner
     | otherwise = do
         cut <- _cut cdb
         t <- newTVarIO mempty
-        m <- initialPayloads cut >>= newTVarIO
+        initialPw <- fmap (PrimedWork . HM.fromList) $
+            forM miners $ \miner ->
+                let mid = view minerId miner
+                in fmap ((mid,) . HM.fromList) $
+                    forM cids $ \cid -> do
+                        let bh = fromMaybe (genesisBlockHeader v cid) (HM.lookup cid (_cutMap cut))
```

Comment:
If some `cid` isn't in `_cutMap cut`, should that be treated as at error?

Reply:
`_cutMap cut` is the blocks in the current cut. When we add new chains, that's done at some future height, so the current cut won't contain those chains when the node starts.